src/components/ApiPublishModal/OpenApiSpec.tsx

```diff
@@ -57,14 +79,84 @@ const OpenApiSpec: React.FC<IProps> = props => {
           },
         ]}
       >
-        <S.Textarea rows={15} placeholder="Enter OpenAPI Spec in YAML/JSON format" />
+        <S.Textarea
+          rows={15}
+          placeholder="Enter OpenAPI Spec in YAML/JSON format"
+          onChange={e => {
+            const spec = e.target.value;
+
+            if (!spec) {
+              setWarnings([]);
+            } else {
+              setWarnings(checkMockingExamples(YAML.parse(JSON.parse(JSON.stringify(spec)))));
+            }
+          }}
+        />
       </Form.Item>
 
+      {renderedWarnings}
+
       <Form.Item name={['mocking', 'enabled']} valuePropName="checked">
         <Checkbox onChange={e => setIsApiMocked(e.target.checked)}>Enable mocking</Checkbox>
       </Form.Item>
     </>
   );
 };
 
+const findResponseExample = (key: string, children: any, check: {hasExample: boolean}) => {
+  if ((key === 'example' && children) || (key === 'examples' && children && Object.entries(children).length)) {
+    check.hasExample = true;
+    return;
+  }
+
+  if (children && typeof children === 'object') {
+    Object.entries(children).forEach(([k, c]) => findResponseExample(k, c, check));
+  }
+};
+
+const checkMockingExamples = (spec: {[key: string]: any}) => {
+  const paths = spec.paths;
+
+  let warnings: string[] = [];
+
+  Object.entries(paths).forEach((pathEntry: [string, any]) => {
+    const [path, pathValue] = pathEntry;
+
+    const pathMocking = pathValue['x-kusk']?.mocking?.enabled;
+
+    if (pathMocking !== false) {
+      Object.entries(pathValue)
```

Comment:
So many steps, I would try to use here single responsibility pattern from SOLID.